source/kernel/src/daemons/shells.rs

```diff
@@ -214,83 +220,103 @@ pub async fn graphical_shell_mono(k: &'static Kernel, settings: GraphicalShellSe
     // Spawn the forth task
     k.spawn(task.run()).await;
 
+    let mut fc_0 = MonoChunk::allocate_mono(FrameLocSize {
+        offset_x: 0,
+        offset_y: char_y,
+        width: disp_width_px,
+        height: disp_height_px - char_y,
+    })
+    .await;
+
+    let mut any = true;
+
+    let interval = Duration::from_secs(1) / 20;
+
     loop {
-        // Wait until there is a frame buffer ready. There wouldn't be if we've spammed frames
-        // before they've been consumed.
-        let mut fc_0 = loop {
-            let fc = disp_hdl
-                .get_framechunk(0, char_y as i32, disp_width_px, disp_height_px - char_y)
-                .await;
-            if let Some(fc) = fc {
-                break fc;
-            } else {
-                k.sleep(Duration::from_millis(10)).await;
-            }
-        };
-        ring_drawer::drawer_bw(&mut fc_0, &rline, style.clone()).unwrap();
-        disp_hdl.draw_framechunk(fc_0).await.unwrap();
+        if any {
+            ring_drawer::drawer_bw(&mut fc_0, &rline, style.clone()).unwrap();
+            fc_0 = disp_hdl.draw_mono(fc_0).await.unwrap();
+            any = false;
+        }
 
-        loop {
-            // loop *just* for skipping released key events
-            futures::select_biased! {
-                event = keyboard.next().fuse() => {
+        let _ = k
+            .timeout(
+                interval,
+                io_poll(&mut any, &mut keyboard, &mut rline, &tid_io),
+            )
+            .await;
```

Comment:
it might be good to have a bigger comment here explaining the debouncing timeout, and why it's okay to cancel `io_poll` abruptly if it times out?

Reply:
Gunna defer on this because I hope to fix it once Instant is a thing. This should be handled in the display IMO and not the shell.